src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -160,6 +163,23 @@ public void undeploy(String id) {
 			.subscribe();
 	}
 
+	public void scale(String id, int desiredCount){
+		Assert.isTrue(desiredCount >= 0, "desiredCount must be greater or equal zero");
+		getStatus(id)
```

Comment:
Why do you need to call getStatus?  Also, if needed, calling the `public AppStatus status(String id)` method would be the way since it takes into account `doOnError` and `block` with a timeout - need to always block with a timeout.

Reply:
I'm following the same pattern used on delete, to see if an app exists before trying to scale it. I forgot about the timeout, fixing that now...